### PR TITLE
Clean up devdocs

### LINKED ISSUE #310: Devdocs are outdated
The `devdocs` folder contains contains multiple documents / files such as
- RFC drafts
  -  mDNS has been stable for some years now we can remove the drafts as they are not used and also should not be implemented.
- RFCs
  - These could be as well removed and replaced by links to the respective documents on the offcial site for RFCs.
The online available versions have the benefit that using these allow direct display of the RFC including the errata.
- Bonjour conformance test
  - This directory holds the files from Apple's "Bonjour Conformance Test" (v1.2.8) distribution.
The files don't include any results and are outdated as the current version is v1.5.4 and could be replaced by a convenience link where to get the "Bonjour Conformance Test" from.

### PR DESCRIPTION
**Pull request by ViToni**
This PR 
- removes unused draft RFCs
- removes existing RFC files
  - keeps links to their online versions including errata
- removes the outdated "Bonjour Conformance Test" distribution files
  - adds link to download location of "Bonjour Conformance Test" (only available with an Apple-ID)

Fixes #310.

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by ViToni** [comment]
@kaikreuzer & @melloware Maybe you could take a look. (@melloware already approved once, but at that time the commits weren't signed off therefore the checks failed. That's fixed now, put the push invalidated the approval.)

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
LGTM

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**
